### PR TITLE
Added Bengali support

### PR DESCRIPTION
**Pull request by harry6396**
|PR checklist|Status|
|-----|:------:|
|Is issue created in GitHub?|[Yes](https://github.com/maqsoftware/Pehla-School-Hindi/issues/224)|
|Are the changes replicated across different locales?|NA|
|Is code analysis task executed and issues resolved in the Java/Kotlin code?|NA|
|Is code analysis task executed and issues resolved in the CPP/native code?|NA|
|Is the version code set in the AndroidManifest.xml or build.gradle file to 1 more than the version which has already been published?|NA|
|Is code commenting done properly for the changes made?|NA|
|Is correct variable naming convention followed?|NA|
|Is auto formatting shortcut Ctrl+Alt+Shift+L executed in all files?|NA|
|Are the message strings used in the application taken from strings.xml file?|NA|
|Are the permissions which are not required removed from AndroidManifest.java file?|NA|
|Is the _uses-feature_ tag used for only must have feature in AndroidManifest.java file?|NA|
|Are the generated files correctly configured in .gitignore?|NA|
|Are the release signing configurations removed from the gradle files?|NA|
|Is commented code removed from all files?|NA|
|Are updated images compressed (< 100 KB)?|NA|
|Ensured there are no typos in any string. For example: Variable/method names, hardcoded strings, commit messages, PR titles, issues, code comments?|Yes|

### CODE REVIEW

File: pehlaschool/Classes/Managers/LanguageManager.cpp [hunk 1]
```diff
@@ -222,6 +222,14 @@ void LanguageManager::initLocalizationMap()
...
+    _localizationMapBengali["Great!"] = "খুব ভালো";
+    _localizationMapSwahili["Great!"] = "Vizuri!";
+
+    _localizationMapEnglish["Are you ready for"] = "Are you ready for"; //
+    // _localizationMapHindi["Are you ready for"] = "क्या तुम तैयार हो";
+    // _localizationMapUrdu["Are you ready for"] = "";
+    _localizationMapBengali["Are you ready for"] = "আপনি পরীক্ষা জন্য প্রস্তুত?";
+
     // _localizationMapBengali["Great!"] = "";
     _localizationMapSwahili["Great!"] = "Vizuri!";
```
Comment (akashdabhi03) [inline pehlaschool/Classes/Managers/LanguageManager.cpp]: Remove this.

File: pehlaschool/Classes/Managers/LanguageManager.cpp [hunk 2]
```diff
@@ -234,463 +242,462 @@ void LanguageManager::initLocalizationMap()
...
     _localizationMapEnglish["NumberPuzzle"] = "Number Blocks";
     _localizationMapHindi["NumberPuzzle"] = "संख्यायों की पहेलियाँ";
     // _localizationMapUrdu["NumberPuzzle"] = "";
-    // _localizationMapBengali["NumberPuzzle"] = "";
+    _localizationMapBengali["NumberPuzzle"] = "সংখ্যা ব্লক";
     _localizationMapSwahili["NumberPuzzle"] = "Fumbo la Nambari";
 
     _localizationMapEnglish["Arrange the numbers in order from smallest to largest"] = "Arrange the numbers in order from smallest to largest.";
-    _localizationMapHindi["Arrange the numbers in order from smallest to largest"] = "छोटी से लेकर बड़ी अंक के क्रम में संख्याओं की क्रमबद्ध करें।";
+    // _localizationMapHindi["Arrange the numbers in order from smallest to largest"] = "छोटी से लेकर बड़ी अंक के क्रम में संख्याओं की क्रमबद्ध करें।";
```
Comment (ankitk20) [inline pehlaschool/Classes/Managers/LanguageManager.cpp]: Uncomment this line

File: pehlaschool/Classes/Menu/DailyScene2.cpp [hunk 3]
```diff
@@ -979,172 +980,172 @@ void DailyScene2::setupFreechoiceTab()
...
+    gquiz1.insert(pair<string, vector<string>>("MultiplicationBoard", {"गुणा पट्ट", "Multiplication Board", "গুণমান ল্যাম্প"}));
+    gquiz1.insert(pair<string, vector<string>>("NumberMatching", {"संख्या मिलाना", "Number Matching", "সংখ্যা ম্যাচিং"}));
+    gquiz1.insert(pair<string, vector<string>>("NumberTracing", {"संख्या अनुरेखण", "Number Tracing", "সংখ্যা ট্রেসিং"}));
+    gquiz1.insert(pair<string, vector<string>>("NumberPuzzle", {"संख्यायों की पहेलियाँ", "Number Puzzle", "সংখ্যা ব্লক"}));
+    gquiz1.insert(pair<string, vector<string>>("NumberTracingExt", {"संख्या अनुरेखण", "Number Tracing", "সংখ্যা ট্রেসিং"}));
+    gquiz1.insert(pair<string, vector<string>>("NumberTrain", {"संख्या ट्रेन", "Number Train", "নম্বর ট্রেন"}));
+    gquiz1.insert(pair<string, vector<string>>("PatternTrain", {"पैटर्न ट्रेन", "Pattern Train", "প্যাটার্ন ট্রেন"}));
+    gquiz1.insert(pair<string, vector<string>>("PlaceValue", {"मान रखे", "Place Value", "স্থানিক মূল্য"}));
+    gquiz1.insert(pair<string, vector<string>>("QuickFacts", {"तुरन्त तथ्य", "Quick Facts", "দ্রুত ঘটনা"}));
+    gquiz1.insert(pair<string, vector<string>>("ReadingBird", {"पढ़ने वाला पक्षी", "ReadingBird", "পাখি পড়া"}));
```
Comment (akashdabhi03) [inline pehlaschool/Classes/Menu/DailyScene2.cpp]: Add space ReadingBird

File: pehlaschool/Classes/Menu/DailyScene2.cpp [hunk 4]
```diff
@@ -979,172 +980,172 @@ void DailyScene2::setupFreechoiceTab()
...
+    gquiz1.insert(pair<string, vector<string>>("SentenceBridge", {"वाक्यों का पुल", "Sentence Bridge", "বাক্য ব্রিজ"}));
+    gquiz1.insert(pair<string, vector<string>>("SentenceMaker", {"वाक्य बनाने वाला", "Sentence Maker", "বাক্য মেকার"}));
+    gquiz1.insert(pair<string, vector<string>>("ShapeMatching", {"आकार मिलाना", "Shape Matching", "আকার ম্যাচিং"}));
+    gquiz1.insert(pair<string, vector<string>>("SoundTrain", {"ध्वनि ट्रेन", "Sound Train", "সাউন্ড ট্রেন"}));
+    gquiz1.insert(pair<string, vector<string>>("Spelling", {"वर्तनी", "Spelling", "বানান"}));
+    gquiz1.insert(pair<string, vector<string>>("Tapping", {"बबल पॉप", "Tapping", "বাবল পপ"}));
+    gquiz1.insert(pair<string, vector<string>>("StarFall", {"टाइपिंग", "Star Fall", "টাইপিং"}));
```
Comment (akashdabhi03) [inline pehlaschool/Classes/Menu/DailyScene2.cpp]: Revert

File: pehlaschool/Classes/Menu/DailyScene2.cpp [hunk 5]
```diff
@@ -979,172 +980,172 @@ void DailyScene2::setupFreechoiceTab()
...
+    gquiz1.insert(pair<string, vector<string>>("PatternTrain", {"पैटर्न ट्रेन", "Pattern Train", "প্যাটার্ন ট্রেন"}));
+    gquiz1.insert(pair<string, vector<string>>("PlaceValue", {"मान रखे", "Place Value", "স্থানিক মূল্য"}));
+    gquiz1.insert(pair<string, vector<string>>("QuickFacts", {"तुरन्त तथ्य", "Quick Facts", "দ্রুত ঘটনা"}));
+    gquiz1.insert(pair<string, vector<string>>("ReadingBird", {"पढ़ने वाला पक्षी", "ReadingBird", "পাখি পড়া"}));
+    gquiz1.insert(pair<string, vector<string>>("SentenceBridge", {"वाक्यों का पुल", "Sentence Bridge", "বাক্য ব্রিজ"}));
+    gquiz1.insert(pair<string, vector<string>>("SentenceMaker", {"वाक्य बनाने वाला", "Sentence Maker", "বাক্য মেকার"}));
+    gquiz1.insert(pair<string, vector<string>>("ShapeMatching", {"आकार मिलाना", "Shape Matching", "আকার ম্যাচিং"}));
+    gquiz1.insert(pair<string, vector<string>>("SoundTrain", {"ध्वनि ट्रेन", "Sound Train", "সাউন্ড ট্রেন"}));
+    gquiz1.insert(pair<string, vector<string>>("Spelling", {"वर्तनी", "Spelling", "বানান"}));
+    gquiz1.insert(pair<string, vector<string>>("Tapping", {"बबल पॉप", "Tapping", "বাবল পপ"}));
```
Comment (akashdabhi03) [inline pehlaschool/Classes/Menu/DailyScene2.cpp]: Revert

File: pehlaschool/Classes/Menu/DailyScene2.cpp [hunk 6]
```diff
@@ -979,172 +980,172 @@ void DailyScene2::setupFreechoiceTab()
...
+    gquiz1.insert(pair<string, vector<string>>("StarFall", {"टाइपिंग", "Star Fall", "টাইপিং"}));
+    gquiz1.insert(pair<string, vector<string>>("ThirtyPuzzle", {"30 पहेली", "Thirty Puzzle", "30 পাজল"}));
+    gquiz1.insert(pair<string, vector<string>>("WhatIsThis", {"यह क्या है?", "What Is This?", "এটা কি?"}));
+    gquiz1.insert(pair<string, vector<string>>("WordTracing", {"शब्द अनुरेखण", "WordTracing", "শব্দ ট্রেসিং"}));
```
Comment (akashdabhi03) [inline pehlaschool/Classes/Menu/DailyScene2.cpp]: Revert

File: pehlaschool/proj.android-studio/app/src/bengali/res/values/array.xml [hunk 7]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (akashdabhi03) [inline pehlaschool/proj.android-studio/app/src/bengali/res/values/array.xml]: This file is not required.

File: pehlaschool/proj.android-studio/app/src/bengali/res/values/colors.xml [hunk 8]
```diff
@@ -0,0 +1,43 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (akashdabhi03) [inline pehlaschool/proj.android-studio/app/src/bengali/res/values/colors.xml]: This file is not required.

File: pehlaschool/proj.android-studio/app/src/bengali/res/values/dimens.xml [hunk 9]
```diff
@@ -0,0 +1,16 @@
+<resources>
```
Comment (akashdabhi03) [inline pehlaschool/proj.android-studio/app/src/bengali/res/values/dimens.xml]: This file is not required.

File: pehlaschool/proj.android-studio/app/src/bengali/res/values/ids.xml [hunk 10]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (akashdabhi03) [inline pehlaschool/proj.android-studio/app/src/bengali/res/values/ids.xml]: This file is not required.

File: pehlaschool/proj.android-studio/app/src/bengali/res/values/strings.xml [hunk 11]
```diff
@@ -0,0 +1,74 @@
+<resources>
+    <string name="app_name">পহলা স্কুল  - Pehla School</string>
```
Comment (akashdabhi03) [inline pehlaschool/proj.android-studio/app/src/bengali/res/values/strings.xml]: Remove double space

File: pehlaschool/proj.android-studio/app/src/bengali/res/values/strings.xml [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+    <string name="content_extraction" translatable="false">এই প্রক্রিয়া শুধুমাত্র একবার সম্পন্ন করা হবে. এটি 10 ​​মিনিট সময় নিতে পারে.</string>
```
Comment (akashdabhi03) [inline pehlaschool/proj.android-studio/app/src/bengali/res/values/strings.xml]: Update fullstop

File: pehlaschool/proj.android-studio/app/src/bengali/res/values/styles.xml [hunk 13]
```diff
@@ -0,0 +1,43 @@
+<resources>
```
Comment (akashdabhi03) [inline pehlaschool/proj.android-studio/app/src/bengali/res/values/styles.xml]: This file is not required.

File: pehlaschool/proj.android-studio/app/src/org/cocos2dx/cpp/VoiceMoldManager.java [hunk 14]
```diff
@@ -64,7 +64,12 @@ public void warmup(String locale) {
     }
 
     public void speak(String text) {
-        speak(text, defaultLocale);
+        if( defaultLocale.equals("bn-IN") && ((!text.equals("")) && (text != null) && (text.matches("^[a-zA-Z0-9_ ?.,'\"]*$")))){
+            speak(text, "hi-IN");
```
Comment (ankitk20) [inline pehlaschool/proj.android-studio/app/src/org/cocos2dx/cpp/VoiceMoldManager.java:125]: Add comment for this line

**Review by akashdabhi03 — CHANGES_REQUESTED** [review]

File: pehlaschool/proj.android-studio/app/src/org/cocos2dx/cpp/VoiceMoldManager.java [hunk 15]
```diff
@@ -64,7 +64,12 @@ public void warmup(String locale) {
     }
 
     public void speak(String text) {
-        speak(text, defaultLocale);
+        if( defaultLocale.equals("bn-IN") && ((!text.equals("")) && (text != null) && (text.matches("^[a-zA-Z0-9_ ?.,'\"]*$")))){
```
Comment (ankitk20) [inline pehlaschool/proj.android-studio/app/src/org/cocos2dx/cpp/VoiceMoldManager.java]: Formatting

**Review by ankitk20 — CHANGES_REQUESTED** [review]

File: pehlaschool/proj.android-studio/app/src/bengali/res/values/dimens.xml (hunk 9 above)
Comment (Shailin) [reply pehlaschool/proj.android-studio/app/src/bengali/res/values/dimens.xml]: Removed

**Review by akashdabhi03 — DISMISSED** [review]

**Review by ankitk20 — APPROVED** [review]

**Review by mathewnik90 — APPROVED** [review]

**Pull request merged**